packages/web/src/components/nav/desktop/PlaylistLibrary.tsx

```diff
@@ -149,10 +188,67 @@ const PlaylistLibrary = ({
     )
   }
 
+  const renderFolder = (folder: PlaylistLibraryFolder) => {
+    return (
+      <PlaylistFolderNavItem
+        key={folder.id}
+        folder={folder}
+        hasUpdate={false}
+        dragging={dragging}
+        draggingKind={draggingKind}
+        onClickEdit={handleClickEditFolder}
+      >
+        {/* {isEmpty(folder.contents) ? null : ( */}
+        <div className={styles.folderContentsContainer}>
+          <LibraryContentsLevel
+            playlists={playlists}
+            contents={[{ playlist_id: 1251, type: 'playlist' }]}
+            renderPlaylist={renderPlaylist}
+            renderExplorePlaylist={renderExplorePlaylist}
+            renderFolder={renderFolder}
+          />
+        </div>
+        {/* )} */}
+      </PlaylistFolderNavItem>
+    )
+  }
+
+  const onClickPlaylist = useCallback(
+    (playlistId: ID, hasUpdate: boolean) => {
+      onClickNavLinkWithAccount()
+      record(
+        make(Name.PLAYLIST_LIBRARY_CLICKED, {
+          playlistId,
+          hasUpdate
+        })
+      )
+    },
+    [record, onClickNavLinkWithAccount]
+  )
+
   // Iterate over playlist library and render out available explore/smart
   // playlists and ordered playlists. Remaining playlists that are unordered
   // are rendered aftewards by sort order.
-  const playlistsNotInLibrary = { ...playlists }
+  const playlistsNotInLibrary = useMemo(() => {
```

Comment:
Would appreciate a closer look at this chunk in particular - I'm not sure what situation this was originally put here to address?

Reply:
If for some reason you have a playlist that is not in your library, we do want to render it in the left nav, so the point of this was to make sure that it shows up (maybe a write failed, etc., we still want to be absolutely sure you see all your playlists). I think this recursion makes sense, just removing from the library as we go. should be very cheap for the normal user. Maybe a function comment explaining that intent would be good. (Or pulling this into a helper w/ the other ones and adding a test case)